pkg/model/api/v1/dashboard.go

```diff
@@ -27,13 +27,23 @@ import (
 type PanelSpec struct {
 	Display common.Display `json:"display" yaml:"display"`
 	Plugin  common.Plugin  `json:"plugin" yaml:"plugin"`
+	Queries []Query  `json:"queries" yaml:"queries"`
 }
 
 type Panel struct {
 	Kind string    `json:"kind" yaml:"kind"`
 	Spec PanelSpec `json:"spec" yaml:"spec"`
 }
 
+type Query struct {
```

Comment:
Did a quick test by adding a new dashboard, adding a TimeSeriesChart panel with an `up` query, and onSave it resulted in: 
<img width="843" alt="image" src="https://user-images.githubusercontent.com/9369625/228343508-c8e99f1c-0fd6-4cc0-893c-317155a7f705.png">

Do you think there will be some additional work on the BE side besides what you mentioned around migration?

Reply:
I am able to save a new dashboard with time series panel with no issue. Are you doing something different than what I did in the video?

https://user-images.githubusercontent.com/9817826/228346184-f7bbb105-f140-4aa4-9884-00b0423eda98.mov

Reply:
Oh wait, duh I need to rebuild the BE again, I’ll try it again in a little bit but I forgot to do that 🙃

Reply:
Okay testing this locally again and yeah it works! Notice a few small things like seeing this warning in the console:

![image](https://user-images.githubusercontent.com/9369625/228850583-37c639fd-1caa-4aa6-8f44-cc6105ff0a5e.png)

Reply:
Having trouble reproducing the can't add a threshold issue now 🤔, the JSON tab not being accurate to where queries moved in panel spec is throwing me off, now I added a few panels, renamed one, and went to save and got:
<img width="663" alt="image" src="https://user-images.githubusercontent.com/9369625/228855825-010fe853-de04-4c12-871e-3986a9c9552f.png">

Here is the JSON I ended up with that wouldn't save:
```json
{ "kind": "Dashboard", "metadata": { "name": "QueriesSpecRefactor", "created_at": "2023-03-28T19:11:01.319754Z", "updated_at": "2023-03-30T13:21:00.259488Z", "version": 6, "project": "perses" }, "spec": { "display": { "name": "QueriesSpecRefactor" }, "panels": { "Test1": { "kind": "Panel", "spec": { "display": { "name": "Test 1" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_duration_seconds" } } } } ], "legend": { "position": "Right" }, "thresholds": { "steps": [ { "value": 0.01 }, { "color": "#FFB249", "value": 0.05 } ] }, "visual": { "line_width": 3, "area_opacity": 0, "point_radius": 2.5, "connect_nulls": false } } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_duration_seconds" } } } } ] } }, "Test1-1": { "kind": "Panel", "spec": { "display": { "name": "Test 1" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "Test1-2": { "kind": "Panel", "spec": { "display": { "name": "Test 1" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "Test1-3": { "kind": "Panel", "spec": { "display": { "name": "Test 1" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "up" } } } } ] } }, "Test1-4": { "kind": "Panel", "spec": { "display": { "name": "Test 1" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_duration_seconds" } } } } ] } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_duration_seconds" } } } } ] } }, "": { "kind": "Panel", "spec": { "display": { "name": "Test" }, "plugin": { "kind": "TimeSeriesChart", "spec": { "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_http_content_length", "series_name_format": "{{ instance }}" } } } } ], "legend": { "position": "Bottom" }, "visual": { "line_width": 1.5, "area_opacity": 0.6, "point_radius": 4, "connect_nulls": true }, "y_axis": { "show": true, "label": "Test", "unit": { "kind": "Decimal", "decimal_places": 2, "abbreviate": false } }, "thresholds": { "steps": [] } } }, "queries": [ { "kind": "TimeSeriesQuery", "spec": { "plugin": { "kind": "PrometheusTimeSeriesQuery", "spec": { "query": "probe_http_content_length", "series_name_format": "{{ instance }}" } } } } ] } } }, "layouts": [ { "kind": "Grid", "spec": { "display": { "title": "Panel Group", "collapse": { "open": true } }, "items": [ { "x": 12, "y": 6, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/Test1" } }, { "x": 12, "y": 0, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/Test1-1" } }, { "x": 0, "y": 0, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/Test1-4" } }, { "x": 0, "y": 6, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/Test1-2" } }, { "x": 0, "y": 12, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/" } } ] } }, { "kind": "Grid", "spec": { "display": { "title": "Second Group", "collapse": { "open": true } }, "items": [ { "x": 0, "y": 0, "width": 12, "height": 6, "content": { "$ref": "#/spec/panels/Test1-3" } } ] } } ], "variables": [], "duration": "5m" } }
```

Reply:
But again, when I start fresh with a brand new dashboard, adding new panels and saving works 😕 😕 

I guess I somehow got one panel into a bad state before and then that happened to be the one I duplicated a bunch before trying to save

Reply:
I'm working on moving the whole panel definition to the JSON tab so ideally that should fix this issue!

Reply:
WIP but most of the work is done on this branch [https://github.com/perses/perses/tree/eunicorn/decouple-query-editor](https://github.com/perses/perses/tree/eunicorn/decouple-query-editor) if you want a sneak peak 👀

Reply:
I cannot reproduce the threshold issue in this branch or [eunicorn/decouple-query-editor](eunicorn/decouple-query-editor) so marking this conversation as resolved! feel free to ping me or comment again if you see another issue :)